sdk/resource/config_test.go

```diff
@@ -0,0 +1,111 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resource_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel/label"
+	opentelemetry "go.opentelemetry.io/otel/sdk"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+func TestDefaultConfig(t *testing.T) {
+	os.Setenv("OTEL_RESOURCE_ATTRIBUTES", "")
```

Comment:
Nit: env variable might potentially affect other tests. Should we reset the env after this case?

BTW, we have an env helper called `EnvStore` in `internal/testing`, and it can record/set/restore env in a convenient way.

Reply:
Fixed. I also updated the other tests in this package to `Restore()` the environment.